feat: Translate docs 2.0 (#868)
ZupIT/beagle-docs: content/pt/ios/customization/dependencies/overview.md

}
```

{{% alert color="success" %}}
Essa estrutura tem um construtor vazio que atribui as implementações padrão do Beagle:

* inicie-o e faça as personalizações necessárias.
{{% /alert %}}

> É importante que se faça essa *configuração* das dependencias durante o processo de inicialização do aplicativo, ou seja, na função `AppDelegate` `didFinishLaunchingWithOptions` conforme mostrado abaixo:

Reviewer: ```suggestion
É importante que se faça essa *configuração* das dependências durante o processo de inicialização da aplicação, ou seja, na função `AppDelegate` `didFinishLaunchingWithOptions` conforme mostrado abaixo:
```

        ))
        dependencies.coder.register(type: CustomComponent.self)
        dependencies.coder.register(type: CustomAction.self)
        BeagleConfigurator.setup(dependencies: dependencies)
        let rootViewController = MainScreen().screenController()
        window?.rootViewController = rootViewController
        return true
    }
}
```

{{% alert color="success" %}}
Para acessar essas dependências durante a execução do aplicativo, o Beagle fornece um wrapper de propriedade `@Injected` que resolve uma instância para o tipo da variável (valor encapsulado).
{{% /alert %}}

### Exemplos

Abaixo temos um exemplo de como limpar o conteúdo do contexto global em algum momento ao executar a aplicação:

```swift
@Injected var globalContext: GlobalContextProtocol
globalContext.clear()
```

{{% alert color="warning" %}}

> Este wrapper de propriedade gera um `fatalError` se for usado para resolver uma dependência que não seja Beagles' ou que seja opcional e não tenha sido configurada, por exemplo:(logger, analyticsProvider, deepLinkHandler, networkClient), para contornar isso, basta usar `@OptionalInjected` que retornará nil se a dependência não for resolvida, conforme mostrado abaixo.

Reviewer: ```suggestion
Este wrapper de propriedade gera um `fatalError` se for usado para resolver uma dependência que não sejado Beagle ou que seja opcional e não tenha sido configurada, por exemplo:(logger, analyticsProvider, deepLinkHandler, networkClient), para contornar isso, basta usar `@OptionalInjected` que retornará nil se a dependência não for resolvida, conforme mostrado abaixo.
```

{{% /alert %}}

```swift
@OptionalInjected var logger: LoggerProtocol?
logger?.log(Log.network(.httpRequest(request: .init(url: urlRequest))))
```

## Dependências customizáveis

Aqui seguem algumas implementações de dependências padrão que podem ser personalizadas.

### CoderProtocol

Ele é responsável pela codificação e decodificação da lógica do Beagle. Ele expõe o método `register` para que seja possível registrar novos componentes e ações.

Abaixo você encontra um exemplo de como registrar um novo componente e uma ação personalizada:

```swift
dependencies.coder.register(component: CustomWidget.self)
dependencies.coder.register(action: CustomAction.self)
```

### UrlBuilderProtocol

Ele é responsável por criar uma URL de requisição do Beagle a partir de URLs relativas que são recebidas de um BFF (nos fluxos de navegação, por exemplo) e uma URL base configurada.

Este é um exemplo de como usar essa configuração:

```swift
dependencies.urlBuilder = UrlBuilder(baseUrl: URL(string: "SUA URL BASE"))
```

### ThemeProtocol

Ele armazena todos os seus **estilos** e sabe como aplicá-los aos seus componentes.

> Alguns widgets possuem uma variável que permite definir o estilo. O nome de cada um deve ser passado para a dependência **Tema**, para que esse estilo possa ser utilizado em seu respectivo componente.

Reviewer: ```suggestion
Alguns widgets possuem uma variável que permite definir o estilo. O nome de cada um deve ser passado para a dependência **Theme**, para que esse estilo possa ser utilizado em seu respectivo componente.
```

Para configurar estilos personalizados, siga estas etapas:

{{% alert color="info" %}}

> Você usará `Text` como um widget que tem `UITextView` como sua **visualização UIKit**. Depois disso, você aplicará suas alterações de estilo a esta visualização, que também terá as personalizações `font` e `textColor`.

Reviewer: ```suggestion
Você usará `Text` como um widget que tem `UITextView` como sua **UIKit view**. Depois disso, você aplicará suas alterações de estilo a esta visualização, que também terá as personalizações `font` e `textColor`.
```

}
```

**Passo 2**: Instancie AppTheme com uma matriz de valores de chave representando o nome do estilo e a função correspondente.

```swift
let theme = AppTheme(styles: [
    "myStyleName": blackTextNormalStyle
])
```

> **Passo 3**: adicione a dependência de tema personalizado às dependências do Beagles

Reviewer: ```suggestion
**Passo 3**: adicione a dependência de tema personalizado às dependências do Beagle
```

### AnalyticsProviderProtocol

> É um protocolo que pode ser implementado para rastrear a aparência da tela, ações e eventos de clique.

Reviewer: ```suggestion
É um protocolo que pode ser implementado para rastrear carregamentos de tela, ações e eventos de clique em componentes Beagle.
```

    func trackEventOnScreenAppeared(_ event: AnalyticsScreen) {
        print("Screen \(event.screenName) appeared")
    }
    func trackEventOnScreenDisappeared(_ event: AnalyticsScreen) {
        print("Screen \(event.screenName) disappeared")
    }
    func trackEventOnClick(_ event: AnalyticsClick) {
        print("Button touch with:\ncategory = \(event.category)\nlabel = \(event.label ?? "empty")\nvalue = \(event.value ?? "empty")")
    }
}
```

## Dependências públicas que não são customizáveis

Aqui listamos algumas dependências que não podem ser customizadas, mas possuem APIs públicas para configurar o Beagle.

### NavigationProtocol

O `navigator` lida com as ações [**navigate**]({{< ref path="/api/actions/navigate/" lang="en" >}}) em seu aplicativo.

> Ele expõe métodos de registro de um controlador de navegação personalizado e uma [** animação de navegação**]({{< ref path="/ios/customization/navigation-animation" lang="en" >}}).

Reviewer: I think that the words which are related to the framework can be kept without translation. e.g: Navigation Controller

> Ele expõe métodos de registro de um controlador de navegação personalizado e uma [** animação de navegação**]({{< ref path="/ios/customization/navigation-animation" lang="en" >}}).

Author: ```suggestion
Ele expõe métodos de registro de um `navigation controler` personalizado e uma [** animação de navegação**]({{< ref path="/ios/customization/navigation-animation" lang="en" >}}).
```
